WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -231,89 +226,94 @@ def post_announcement
   end
 
   def post_results
-    I18n.with_locale :en do
-      comp = Competition.find(params[:id])
-      unless comp.results
-        return render html: "<div class='container'><div class='alert alert-warning'>No results</div></div>".html_safe
-      end
+    if ComputeAuxiliaryData.in_progress?
+      flash[:warning] = "Please wait until auxiliary data is computed."
+      redirect_to admin_edit_competition_path(competition_from_params)
```

Comment:
I'd probably rewrite this line as `return redirect_to admin_edit_competition_path(competition_from_params)`, so the rest of the method could remain unchanged.